dds/DCPS/XTypes/TypeObject.cpp

```diff
@@ -471,616 +471,719 @@ const TypeMap TypeMapBuilder::EmptyMap;
+                              TypeIdentifierSet& dependencies,
+                              TypeIdentifierSet& sccs)
   {
-    compute_dependencies_i(type_map, type.header, dependencies);
-    compute_dependencies_i(type_map, *type.element_identifier, dependencies);
+    compute_dependencies_i(type_map, type.header, dependencies, sccs);
+    compute_dependencies_i(type_map, *type.element_identifier, dependencies, sccs);
   }
 
   void compute_dependencies_i(const TypeMap& type_map,
                               const PlainMapSTypeDefn& type,
-                              OPENDDS_SET(TypeIdentifier)& dependencies)
+                              TypeIdentifierSet& dependencies,
+                              TypeIdentifierSet& sccs)
   {
-    compute_dependencies_i(type_map, type.header, dependencies);
-    compute_dependencies_i(type_map, *type.element_identifier, dependencies);
-    compute_dependencies_i(type_map, *type.key_identifier, dependencies);
+    compute_dependencies_i(type_map, type.header, dependencies, sccs);
+    compute_dependencies_i(type_map, *type.element_identifier, dependencies, sccs);
+    compute_dependencies_i(type_map, *type.key_identifier, dependencies, sccs);
   }
 
   void compute_dependencies_i(const TypeMap& type_map,
                               const PlainMapLTypeDefn& type,
-                              OPENDDS_SET(TypeIdentifier)& dependencies)
-  {
-    compute_dependencies_i(type_map, type.header, dependencies);
-    compute_dependencies_i(type_map, *type.element_identifier, dependencies);
-    compute_dependencies_i(type_map, *type.key_identifier, dependencies);
-  }
-
-  void compute_dependencies_i(const TypeMap&,
-                              const StronglyConnectedComponentId&,
-                              OPENDDS_SET(TypeIdentifier)&)
+                              TypeIdentifierSet& dependencies,
+                              TypeIdentifierSet& sccs)
   {
-    // Do nothing.
+    compute_dependencies_i(type_map, type.header, dependencies, sccs);
+    compute_dependencies_i(type_map, *type.element_identifier, dependencies, sccs);
+    compute_dependencies_i(type_map, *type.key_identifier, dependencies, sccs);
   }
 
   template <typename T>
   void compute_dependencies_i(const TypeMap& type_map,
                               const Sequence<T>& type,
-                              OPENDDS_SET(TypeIdentifier)& dependencies)
+                              TypeIdentifierSet& dependencies,
+                              TypeIdentifierSet& sccs)
   {
     for (typename Sequence<T>::const_iterator pos = type.begin(), limit = type.end(); pos != limit; ++pos) {
-      compute_dependencies_i(type_map, *pos, dependencies);
+      compute_dependencies_i(type_map, *pos, dependencies, sccs);
+    }
+  }
+
+  TypeIdentifier make_scc_id(const TypeIdentifier& tid)
+  {
+    if (tid.kind() != TI_STRONGLY_CONNECTED_COMPONENT) {
+      return tid;
```

Comment:
Should this return an error instead? It is returning the input type which isn't an SCC which can be misleading given the function name `make_scc_id`.

Reply:
The function does exactly what I want it to do, return the SCCId if it's a component of an SCC or the given type identifier if not.  Would `make_scc_id_or_default` be better?

Reply:
Yeah, I think so.